ports/stm/common-hal/sdioio/SDCard.c

```diff
@@ -0,0 +1,356 @@
+        const mcu_periph_obj_t *mcu_sdio_data0 = NULL;
+        if(!(mcu_sdio_data0 = find_pin_function(mcu_sdio_data0_list, sdio_data0_len, data[0], periph_index))) {
+            continue;
+        }
+
+        const mcu_periph_obj_t *mcu_sdio_data1 = NULL;
+        if(num_data > 1 && !(mcu_sdio_data1 = find_pin_function(mcu_sdio_data1_list, sdio_data1_len, data[1], periph_index))) {
+            continue;
+        }
+
+        const mcu_periph_obj_t *mcu_sdio_data2 = NULL;
+        if(num_data > 2 && !(mcu_sdio_data2 = find_pin_function(mcu_sdio_data2_list, sdio_data2_len, data[2], periph_index))) {
+            continue;
+        }
+
+        const mcu_periph_obj_t *mcu_sdio_data3 = NULL;
+        if(num_data > 3 && !(mcu_sdio_data3 = find_pin_function(mcu_sdio_data3_list, sdio_data3_len, data[3], periph_index))) {
+            continue;
+        }
+
+        if (reserved_sdio[periph_index-1]) {
+            sdio_taken = true;
+            continue;
+        }
+
+        self->clock = mcu_sdio_clock;
+        self->command = mcu_sdio_command;
+        self->data[0] = mcu_sdio_data0;
+        self->data[1] = mcu_sdio_data1;
+        self->data[2] = mcu_sdio_data2;
+        self->data[3] = mcu_sdio_data3;
+
+        return periph_index;
+    }
+
+    if (sdio_taken) {
+        mp_raise_ValueError(translate("Hardware busy, try alternative pins"));
+    } else {
+        mp_raise_ValueError_varg(translate("Invalid %q pin selection"), MP_QSTR_SDIO);
+    }
+}
+
+
+void common_hal_sdioio_sdcard_construct(sdioio_sdcard_obj_t *self,
+        const mcu_pin_obj_t * clock, const mcu_pin_obj_t * command,
+        uint8_t num_data, mcu_pin_obj_t ** data, uint32_t frequency) {
+
+    int periph_index = check_pins(self, clock, command, num_data, data);
+    SDIO_TypeDef * SDIOx = mcu_sdio_banks[periph_index - 1];
+
+    GPIO_InitTypeDef GPIO_InitStruct = {0};
+
+    // /* GPIOC and GPIOD Periph clock enable */
+    // RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOC | RCC_AHB1Periph_GPIOD | SD_DETECT_GPIO_CLK, ENABLE);
+
+    /* Configure data PC.08, PC.09, PC.10, PC.11 pins: D0, D1, D2, D3 pins */
+    for (int i=0; i<num_data; i++) {
+        GPIO_InitStruct.Pin = pin_mask(data[i]->number);
+        GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+        GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+        GPIO_InitStruct.Pull = GPIO_PULLUP;
+        GPIO_InitStruct.Alternate = self->data[i]->altfn_index;
+        HAL_GPIO_Init(pin_port(data[i]->port), &GPIO_InitStruct);
+    }
+
+    /* Configure PD.02 CMD line */
+    GPIO_InitStruct.Alternate = self->command->altfn_index;
+    GPIO_InitStruct.Pin = pin_mask(command->number);
+    HAL_GPIO_Init(pin_port(command->port), &GPIO_InitStruct);
+
+    /* Configure PC.12 pin: CLK pin */
+    GPIO_InitStruct.Alternate = self->clock->altfn_index;
+    GPIO_InitStruct.Pin = pin_mask(clock->number);
+    HAL_GPIO_Init(pin_port(clock->port), &GPIO_InitStruct);
+
+// XXX hard coded pin
+#define SD_DETECT_PIN GPIO_PIN_12
```

Comment:
Can we move this to the board definition? Or is it hard-coded? PB_12 is a widely used pin.

Reply:
removed.